src/stlink-lib/usb.c

```diff
@@ -1221,6 +1221,8 @@ stlink_t *stlink_open_usb(enum ugly_loglevel verbose, enum connect_type connect,
 
     libusb_free_device_list(list, 1);
 
+// libusb_kernel_driver_active is not available on Windows.
+#if !defined(_WIN32)
```

Comment:
Theoretically, this can be omitted. The function will simply return LIBUSB_ERROR_NOT_SUPPORTED.
https://github.com/libusb/libusb/blob/1170db79bdc4155c3d5a64787fcc4f8d56b26174/libusb/core.c#L1974-L1989

But let it be a less of unnecessary code)

Reply:
yeah, just leave it, it's ok - we have way more serious issues to deal with in other parts of the codebase.

Reply:
I’m in favor of handling `LIBUSB_ERROR_NOT_SUPPORTED` instead of adding this `_WIN32` check.

Reply:
@gszy Ok, go for it then... I don't mind.

Reply:
Well, it seems no change is necessary: we only check if `libusb_kernel_driver_active() == 1`, so if the function returned  `LIBUSB_ERROR_NOT_SUPPORTED`, the program would continue in the same way it would do if the function returned `0` (or anything other than `1`). I don’t think we have to check if we’re on WIN32, since libusb already does it.

It may be useful to handle other return values properly, like `LIBUSB_ERROR_INVALID_PARAM`, though it may be out of this PR’s scope.

I understand @hydroconstructor’s change could somewhat reduce stlink binary size on Windows, but I don’t think these additional two lines of code are worth the benefit.

Reply:
@hydroconstructor Let's remove this check.